lib/voxpupuli/test/rake.rb

```diff
@@ -1,36 +1,44 @@
 # frozen_string_literal: true
 
-require 'puppetlabs_spec_helper/rake_tasks'
+require 'metadata-json-lint/rake_task'
+require 'puppet-strings/tasks'
+require 'rspec/core/rake_task'
+
+begin
+  require 'parallel_tests'
+  parallel_tests_loaded = true
+rescue LoadError
+  parallel_tests_loaded = false
+end
 
-PuppetLint.configuration.log_format = '%<path>s:%<line>s:%<check>s:%<KIND>s:%<message>s'
-# without this, puppet-lint always gives an exit code of 0
-PuppetLint.configuration.fail_on_warnings = true
+# load and configure puppet-lint
+require_relative 'rake/puppetlint'
 
-desc 'Run tests'
-task test: [:release_checks]
+# load and configure puppet-syntax
+require_relative 'rake/puppetsyntax'
 
-task default: [:release_checks]
+# define various helper tasks
+require_relative 'rake/check'
+
+# define the various spec tasks
+require_relative 'rake/spec'
 
-namespace :check do
-  desc 'Check for trailing whitespace'
-  task :trailing_whitespace do
-    errors = []
-
-    Dir.glob('**/*.md', File::FNM_DOTMATCH).sort.each do |filename|
-      next if filename =~ %r{^((modules|acceptance|\.?vendor|spec/fixtures|pkg)/|REFERENCE.md)}
-
-      File.foreach(filename).each_with_index do |line, index|
-        if line =~ /\s\n$/
-          errors << "#{filename} has trailing whitespace on line #{index + 1}"
-        end
-      end
-    end
-
-    if errors.any?
-      errors.each { |error| puts error }
-      exit 1
-    end
+# static code analysis
+require_relative 'rake/validate'
+
+desc 'Runs all necessary checks on a module in preparation for a release'
+task :release_checks do
+  Rake::Task[:lint].invoke
+  Rake::Task[:validate].invoke
+  if parallel_tests_loaded
+    Rake::Task[:parallel_spec].invoke
+  else
+    Rake::Task[:spec].invoke
   end
+  Rake::Task[:check].invoke
 end
```

Comment:
Since we made `parallel_tests` a runtime dependency I think this can be an trivial task that depends on others:
```suggestion
task release_checks: [:lint, :validate, :parallel_spec, :check]
```

Reply:
should we keep the release_checks task? I just renamed it to test

Reply:
discussed on IRC, we will drop the unused `release_checks` task.

Reply:
I'm fine renaming it to `test`. It's also in line with the gem name (`voxpupuli-test`).